# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/nsIExternalProtocolService.idl
# Commit: f0094c9184c4
# Full Hash: f0094c9184c41675f11efc2d4332ec1a63e570bc
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2019-11-27 09:44:05
# Regressor Bug: 1496380
# File Overlap Count: 1
# Description:
#   Bug 1496380 - stop recursion via the external protocol handler if Firefox is either the default OS handler or a configured external handler, r=mossop
#   
#   This is an initial implementation of this idea that works on mac.
#   I've added a Windows implementation in another commit in this stack. I'll
#   look at a Linux one in a follow-up bug. I do not think we need them in the
# ==============================================================================

diff -r 9dbf3a2c8c37 -r f0094c9184c4 uriloader/exthandler/nsIExternalProtocolService.idl
--- a/uriloader/exthandler/nsIExternalProtocolService.idl	Tue Nov 26 18:47:20 2019 +0000
+++ b/uriloader/exthandler/nsIExternalProtocolService.idl	Tue Nov 26 18:51:04 2019 +0000
@@ -123,4 +123,11 @@
    *        possible to get a description for it.
    */
   AString getApplicationDescription(in AUTF8String aScheme);
+
+  /**
+   * Check if this app is registered as the OS default for a given scheme.
+   *
+   * @param aScheme The scheme to look up. For example, "mms".
+   */
+  bool isCurrentAppOSDefaultForProtocol(in AUTF8String aScheme);
 };